src/ethereum_spec_tools/new_fork/cli.py

```diff
@@ -0,0 +1,216 @@
+"""
+Command-line interface to `ForkBuilder`.
+"""
+
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Sequence
+
+from ethereum_types.numeric import U64, Uint
+
+from ethereum.fork_criteria import ByBlockNumber, ByTimestamp, Unscheduled
+
+from ..forks import Hardfork
+from .builder import ForkBuilder, SetConstant
+
+
+def _make_parser() -> ArgumentParser:
+    forks = Hardfork.discover()
+    fork_short_names = [f.short_name for f in forks]
+
+    parser = ArgumentParser(
+        # TODO: Description / help text
+    )
+
+    parser.add_argument(
+        "--template-fork",
+        "--from_fork",
+        dest="template_fork",
+        type=str,
+        choices=fork_short_names,
+        metavar="NAME",
+        help="short name of the fork to use as a template",
+        default=fork_short_names[-1],
+    )
+
+    parser.add_argument(
+        "--new-fork",
+        "--to_fork",
+        dest="new_fork",
+        type=str,
+        required=True,
+        metavar="NAME",
+        help="short name (Python-friendly) of the new fork",
+    )
+
+    parser.add_argument(
+        "--output",
+        "-o",
+        dest="output",
+        type=Path,
+        metavar="PATH",
+        help="directory in which to place the generated fork",
+    )
+
+    parser.add_argument(
+        "--force",
+        "-f",
+        action="store_true",
+        help="overwrite the destination if it already exists",
+        default=False,
+    )
+
+    fork_criteria = parser.add_mutually_exclusive_group()
+
+    fork_criteria.add_argument(
+        "--unscheduled",
+        action="store_const",
+        const=Unscheduled(),
+        dest="fork_criteria",
+        help="mark the new fork as unscheduled",
+    )
+
+    fork_criteria.add_argument(
+        "--at-timestamp",
+        type=lambda x: ByTimestamp(int(x)),
+        dest="fork_criteria",
+        help="mark the new fork as beginning at the given time",
+    )
+
+    fork_criteria.add_argument(
+        "--at-block",
+        type=lambda x: ByBlockNumber(int(x)),
+        dest="fork_criteria",
+        help="mark the new fork as beginning at the given block number",
+    )
+
+    blob_parameters = parser.add_argument_group()
+
+    blob_parameters.add_argument(
```

Comment:
This could get a bit more complicated since post-Osaka, we won't have TARGET_BLOB_GAS_PER_BLOCK as ain independent constant. See [this](https://github.com/ethereum/execution-specs/pull/1370)

Reply:
Should we disable this option post-osaka or allow users to modify it at their own risk?